include/boost/math/ccmath/isinf.hpp

```diff
@@ -6,16 +6,26 @@
 #ifndef BOOST_MATH_CCMATH_ISINF
 #define BOOST_MATH_CCMATH_ISINF
 
+#include <cmath>
 #include <limits>
+#include <boost/math/tools/is_constant_evaluated.hpp>
 
-namespace boost { namespace math { namespace ccmath {
+namespace boost::math::ccmath {
 
 template <typename T>
 inline constexpr bool isinf(T x)
 {
-    return x == std::numeric_limits<T>::infinity() || -x == std::numeric_limits<T>::infinity();
+    if(BOOST_MATH_IS_CONSTANT_EVALUATED(x))
```

Comment:
Shouldn't this be an `if constexpr`?

Reply:
I thought the same thing. Here is the [SO answer.](https://stackoverflow.com/questions/54251530/stdis-constant-evaluated-behavior)

Reply:
Yeah, it's a weird corner of the language for sure, but it must absolutely not be constexpr!

Reply:
Wowsers, well good thing we have some real programmers eyes on this one! I would've Pied Piper'd us on this one. . .